src/ray/raylet/local_object_manager.cc

```diff
@@ -180,8 +193,8 @@ void LocalObjectManager::SpillObjectsInternal(
         io_worker->rpc_client()->SpillObjects(
             request, [this, objects_to_spill, callback, io_worker](
                          const ray::Status &status, const rpc::SpillObjectsReply &r) {
-              io_worker_pool_.PushSpillWorker(io_worker);
               absl::MutexLock lock(&mutex_);
+              io_worker_pool_.PushSpillWorker(io_worker);
```

Comment:
Is this a thread-safety bug due to io_worker_pool_ missing a GUARDED_BY annotation?

Reply:
Yeah. I didn't really find an issue, but when I looked at `PushSpillWorker `, it wasn't thread safe, so I moved it here.